test/e2e/priority_scheduling/priority_scheduling_test.go

```diff
@@ -251,6 +252,217 @@ var _ = ginkgo.Describe("PriorityScheduling", func() {
 	})
 
+	ginkgo.It("Verify_Gang_Scheduling_With_Priority", func() {
+		By("Setting custom YuniKorn configuration")
+		annotation = "ann-" + common.RandSeq(10)
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "fifo", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:      "default",
+				Parent:    false,
+				Resources: configs.Resources{Max: map[string]string{siCommon.CPU: "100m", siCommon.Memory: "100M"}},
+			}); err != nil {
+				return err
+			}
+
+			return nil
+		})
+		sleepPodConf = k8s.TestPodConfig{
+			Name: "test-sleep-" + common.RandSeq(5),
+			Labels: map[string]string{
+				constants.LabelQueueName:     "root.default",
+				constants.LabelApplicationID: "app-sleep-" + common.RandSeq(5)},
+			Namespace: ns,
+			Resources: rr,
+		}
+
+		taskGroupMinResource := map[string]resource.Quantity{}
+		for k, v := range rr.Requests {
+			taskGroupMinResource[k.String()] = v
+		}
+		lowPodConf = k8s.TestPodConfig{
+			Name: "test-low-priority-" + common.RandSeq(5),
+			Labels: map[string]string{
+				constants.LabelQueueName:     "root.default",
+				constants.LabelApplicationID: "app-low-" + common.RandSeq(5)},
+			Namespace: ns,
+			Annotations: &k8s.PodAnnotation{
+				TaskGroupName: "group-low",
+				TaskGroups: []interfaces.TaskGroup{
+					{
+						Name:        "group-low",
+						MinMember:   int32(1),
+						MinResource: taskGroupMinResource,
+					},
+				},
+			},
+			Resources:         rr,
+			PriorityClassName: lowPriorityClass.Name,
+			RestartPolicy:     v1.RestartPolicyNever,
+		}
+
+		normalPodConf = k8s.TestPodConfig{
+			Name: "test-normal-priority-" + common.RandSeq(5),
+			Labels: map[string]string{
+				constants.LabelQueueName:     "root.default",
+				constants.LabelApplicationID: "app-normal-" + common.RandSeq(5)},
+			Namespace: ns,
+			Annotations: &k8s.PodAnnotation{
+				TaskGroupName: "group-normal",
+				TaskGroups: []interfaces.TaskGroup{
+					{
+						Name:        "group-normal",
+						MinMember:   int32(1),
+						MinResource: taskGroupMinResource,
+					},
+				},
+			},
+			Resources:         rr,
+			PriorityClassName: normalPriorityClass.Name,
+			RestartPolicy:     v1.RestartPolicyNever,
+		}
+
+		highPodConf = k8s.TestPodConfig{
+			Name: "test-high-priority-" + common.RandSeq(5),
+			Labels: map[string]string{
+				constants.LabelQueueName:     "root.default",
+				constants.LabelApplicationID: "app-high-" + common.RandSeq(5)},
+			Namespace: ns,
+			Annotations: &k8s.PodAnnotation{
+				TaskGroupName: "group-high",
+				TaskGroups: []interfaces.TaskGroup{
+					{
+						Name:        "group-high",
+						MinMember:   int32(1),
+						MinResource: taskGroupMinResource,
+					},
+				},
+			},
+			Resources:         rr,
+			PriorityClassName: highPriorityClass.Name,
+			RestartPolicy:     v1.RestartPolicyNever,
+		}
```

Comment:
See if you can make the whole code starting from line no.285 simpler by adding a new method to return the test pod config for all high, low etc